pkg/controller/tektoninstallation/tektoninstallation_controller.go

```diff
@@ -87,37 +105,111 @@ func (r *ReconcileTektonInstallation) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
-	err := r.EnsureTektonSubscription(reqLogger, tektonInstallation)
-	return reconcile.Result{}, err
-}
+	if created, err := r.ensureTektonSubscription(reqLogger, tektonInstallation, SubscriptionNamespace); err != nil {
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, tektonInstallation, r.setStatusTektonSubscriptionFailed, err, "failed to create tekton subscription in namespace %s", SubscriptionNamespace)
+	} else if created {
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstalling, "created tekton subscription")
+	}
 
-// EnsureTektonSubscription ensures that there is an OLM Subscription resource for Tekton
-func (r *ReconcileTektonInstallation) EnsureTektonSubscription(logger logr.Logger, tektonInstallation *v1alpha1.TektonInstallation) error {
-	tektonSubNamespace := SubscriptionNamespace
-	if err := r.ensureTektonSubscription(logger, tektonInstallation, tektonSubNamespace); err != nil {
-		return r.wrapErrorWithStatusUpdate(logger, tektonInstallation, r.setStatusTektonSubscriptionFailed, err, "failed to create tekton subscription in namespace %s", tektonSubNamespace)
+	if requeue, err := r.ensureWatchTektonConfig(); err != nil {
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationFailed, err, "failed to get TektonConfig")
+	} else if requeue {
+		return reconcile.Result{Requeue: true, RequeueAfter: 3 * time.Second}, nil
+	}
+
+	tektonCfg := &config.Config{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: TektonConfigName}, tektonCfg)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstalling, err, "TektonConfig is installing")
+		}
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationFailed, err, "failed to get TektonConfig")
+	}
+
+	code, details := getTektonConfigStatus(tektonCfg)
+	switch code {
+	case config.InstalledStatus:
+		reqLogger.Info("done with Tekton installation")
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationSucceeded, "tekton installation succeeded")
```

Comment:
we don't need the message here
```suggestion
		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationSucceeded, "")
```